library/network/src/modules/SuSEFirewall.rb

```diff
@@ -403,6 +405,255 @@ def _sf2_to_firewalld_service(service)
+        end
+        # Configure the zones
+        GetKnownFirewallZones().each do |zone|
+          if @SETTINGS[zone][:modified].empty?
+            Builtins.y2milestone("zone=#{zone} hasn't been modified. Skipping...")
+            next
+          end
+          # Drop duplicates
+          @SETTINGS[zone][:modified].uniq!
+          if @SETTINGS[zone][:modified].include?(:masquerade)
+            @SETTINGS[zone][:masquerade] ? @fwd_api.add_masquerade(zone) :
+              @fwd_api.remove_masquerade(zone)
+            @SETTINGS[zone][:modified].delete(:masquerade)
+          end
+
+          if @SETTINGS[zone][:modified].include?(:interfaces)
+            # These are the ones which should be enabled
+            good_interfaces = @SETTINGS[zone][:interfaces]
+            # These are the ones which are enabled
+            current_interfaces = @fwd_api.get_interfaces_in_zone(zone)
+            # And these are the ones which should be dropped
+            to_drop = current_interfaces - good_interfaces
+            to_add = good_interfaces - current_interfaces
+            Builtins.y2debug("Interfaces: drop -> #{to_drop} add -> #{to_add}")
+            to_drop.each { |rmif| @fwd_api.remove_interface_from_zone(zone, rmif) }
+            to_add.each { |gif| @fwd_api.add_interface_to_zone(zone, gif) }
+            @SETTINGS[zone][:modified].delete(:interfaces)
+          end
+
+          if @SETTINGS[zone][:modified].include?(:services)
+            # These are the services which should be enabled
+            good_services = @SETTINGS[zone][:services]
+            # These are the ones which are enabled
+            current_services = @fwd_api.get_services_in_zone(zone)
+            # And these are the ones which should be dropped
+            to_drop = current_services - good_services
+            to_add = good_services - current_services
+            Builtins.y2debug("Services: drop -> #{to_drop} add -> #{to_add}")
+            to_drop.each { |rms| @fwd_api.remove_service_from_zone(zone, rms) }
+            to_add.each { |gs| @fwd_api.add_service_to_zone(zone, gs) }
+            @SETTINGS[zone][:modified].delete(:services)
+          end
+
+          if @SETTINGS[zone][:modified].include?(:ports)
+            # These are the ports which should be enabled
+            good_ports = @SETTINGS[zone][:ports]
+            # These are the ones which are enabled
+            current_ports = @fwd_api.get_ports_in_zone(zone)
+            # And these are the ones which should be dropped
+            to_drop = current_ports - good_ports
+            to_add = good_ports - current_ports
+            Builtins.y2debug("Ports: drop -> #{to_drop} add -> #{to_add}")
+            to_drop.each { |rmp| @fwd_api.remove_port_from_zone(zone, rmp) }
+            to_add.each { |gp| @fwd_api.add_port_to_zone(zone, gp) }
+            @SETTINGS[zone][:modified].delete(:ports)
+          end
+
+          if @SETTINGS[zone][:modified].include?(:protocols)
```

Comment:
Yes I just added it there for consistency since we may need it in the future so I made this method as complete as possible. I think I will have to add it to the SetAllowedServicesForZoneProto sometime in the near future.